platform/lib/utils/limitedRemoteFetch.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2020 The AMPHTML Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fetch = require('node-fetch');
+const url = require('url');
+const LRU = require('lru-cache');
+const config = require('@lib/config.js');
+const RemoteFetchError = require('./remoteFetchError');
+
+/**
+ * The time in milliseconds in which HOST_RATE_LIMIT requests can
+ * happen before the user needs to wait
+ * @type {Number}
+ */
+const RATE_LIMIT_TIME_FRAME = 10 * 1000;
+
+/**
+ * The number of times a certain host can be requested in RATE_LIMIT_TIME_FRAME
+ * @type {Number}
+ */
+const HOST_RATE_LIMIT = 10;
+
+/**
+ * The maximum count of limits that is tracked.
+ * @type {Number}
+ */
+const MAX_LIMITS = 500;
+
+const limits = new LRU({
+  max: MAX_LIMITS,
+  maxAge: RATE_LIMIT_TIME_FRAME,
+});
+
+class LimitedRemoteFetch {
+  constructor({requestHeaders}) {
+    this.requestHeaders = requestHeaders || {};
+  }
+
+  /**
+   * Fetches a user-defined remote URL and returns the response.
+   * Verifies that the returned response is a proper HTML document.
+   *
+   * @param  {String} urlString
+   * @return {Response}
+   */
+  async fetchHtmlResponse(urlString) {
```

Comment:
Please rename to `fetchHtml` (the response is implied) as part of the extraction.

Reply:
`fetchHtml` would sound like it returns html. Html is in the name to make it clear that only `text/html` is accepted.
Maybe when we turn it around it would not sound so much like it returns html `htmlFetch`?
Would we also want to rename `fetchDocument` since it does not return a response like `fetch` does?
Maybe `loadContent`?

Reply:
Hm, makes sense. Then lets use `fetchHtmlResponse` and `fetchHtmlDocument` for consistency.